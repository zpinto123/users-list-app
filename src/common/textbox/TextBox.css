.textBox {
  height: 25px;
  font-size: 16px;
}