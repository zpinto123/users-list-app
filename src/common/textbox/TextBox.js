import React from "react";
import PropTypes from 'prop-types';

import './TextBox.css';

const TextBox = ({ placeholder, handleChange, type }) => (
    <input className="textBox" type={type} onChange={handleChange} placeholder={placeholder}/>
);

TextBox.propTypes = {
  type: PropTypes.string,
  placeholder: PropTypes.string,
  handleChange: PropTypes.func.isRequired
}

TextBox.defaultProps = {
  type: "text",
  placeholder: "",
}

export default TextBox;

