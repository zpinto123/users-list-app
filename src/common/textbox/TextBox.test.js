import React from "react";

import TextBox from "./TextBox";

import renderer from "react-test-renderer";
describe("<TextBox />", () => {
  it("renders correctly", () => {
    const mockCallback = jest.fn();
    const component = renderer
      .create(<TextBox handleChange={mockCallback} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
