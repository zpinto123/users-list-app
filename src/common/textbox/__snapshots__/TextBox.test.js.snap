// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextBox /> renders correctly 1`] = `
<input
  className="textBox"
  onChange={[MockFunction]}
  placeholder=""
  type="text"
/>
`;
